### ISSUE TITLE
New tab (`Ctrl+T`) - text typed in the address bar disappears

### ISSUE THREAD

**Issue by maricn** [tridactyl/tridactyl#2840]
Tridactyl version: 1.20.1pre4591-6d0810f2
Firefox version: Mozilla Firefox 81.0
Operating system: linux

*   Brief description of the problem:

After opening a new page with Firefox's `Ctrl+T` and start typing in the URL bar, the URL bar content resets when the tridactyl's blank page is loaded. Usually this cuts out the first part of the typed string, and can be pretty annoying especially when relying on the results from Firefox history.

*   Steps to reproduce:

1. open new tab with `Ctrl+T`
2. immediately start typing something
3. observe that the typed text has disappeared (or at least the beginning part typed before the page loaded)

*   URL of the website the bug happens on:
New tab

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com):
https://gist.github.com/maricn/a86720274072a9899d747524e8aacd84 (rev. 5)

*   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist):

```
" don't sanitize https://github.com/tridactyl/tridactyl/issues/1409
" sanitise tridactyllocal tridactylsync

" General Settings
set configversion 1.9
set update.lastchecktime 1593077346454
set smoothscroll true
set modeindicatorshowkeys true
set modeindicator false

" Theme
" set theme maricn-base16-gruvbox-dark-hard
" set colours maricn-base16-gruvbox-dark-hard
colourscheme maricn-base16-gruvbox-dark-hard

" Binds
unbind <F1>
" url increment/decrement
unbind <C-a>
unbind <C-x>
" scrolls
unbind <C-d>
unbind <C-u>
unbind <C-f>
unbind <C-b>
" clipboard tabopen
unbind p
unbind P
" repeat
unbind .
" unbind s
" unbind S
" unbind D
" unbind U

" bind d scrollpage 0.5
" bind u scrollpage -0.5

" conflicts with <C-d> as "Bookmark this page"
" bind <C-d> scrollpage 0.7
" bind <C-u> scrollpage -0.7

bind <C-[> back
bind <C-]> forward

bind i mode ignore
bind --mode=ignore <escape> mode normal

bind t tabnew
bind J tabnext
bind K tabprev
bind x tabclose
bind --mode=ignore ^ tab #

bind / fillcmdline find
bind n findnext
bind N findnext -1
bind <Escape> composite mode normal ; hidecmdline ; nohlsearch

bind --mode=ex <ArrowRight> ex.insert_completion

" For syntax highlighting see https://github.com/tridactyl/vim-tridactyl
" vim: set filetype=tridactyl
```

**Comment by bovine3dom** [comment]
I can't reproduce this (FF81, Linux). Can anyone else?

It shouldn't really be something we can do. I can't reproduce it even if I `set newtab news.bbc.co.uk`, which is what I thought might be causing it.

**Comment by maricn** [comment]
Actually, if you're a real ninja on the keyboard, you can achieve it also with `t`, `Ctrl+L` (or `F6`), `asdfasdfafds` (`:tabnew`, focus URL bar, start typing). Or at least on my machine..

**Comment by maricn** [comment]
I tried it out as far back as 1.16.2 release and it's present all the way there...

It looks more like when the extension loads Firefox or Tridactyl reset the URL bar content. If it's in Tridactyl control, then we should grab what is already there when the extension loads. Otherwise, it might be we can't do anything about it.

**Comment by bovine3dom** [comment]
I can't reproduce this and I strongly suspect it isn't our fault. Perhaps you could file an issue with bugzilla?

I'm inclined to suspect that it might be their "your new tab page has changed" pop-up which is causing the issues.

**Comment by maricn** [comment]
> I can't reproduce this and I strongly suspect it isn't our fault. Perhaps you could file an issue with bugzilla?

Makes sense, thanks for looking into it.

> I'm inclined to suspect that it might be their "your new tab page has changed" pop-up which is causing the issues.

It's not due to that pop-up, it's during the usual use... When pop-up appears, URL bar doesn't receive focus at all. Btw, it would also be great if the pop-up could be silenced somehow :)

**Comment by maricn** [comment]
> I can't reproduce this and I strongly suspect it isn't our fault. Perhaps you could file an issue with bugzilla?

Just found one 3y old issue that seems to be related to this problem. Just attaching a cross-reference here to https://bugzilla.mozilla.org/show_bug.cgi?id=1427535
:)

**Comment by bovine3dom** [comment]
> Btw, it would also be great if the pop-up could be silenced somehow :)

In theory it should be possible with userChrome.css but I haven't spent very long thinking about it.

**Comment by jamesnvc** [comment]
I've been experiencing this issue recently as well; glad to see it's (probably) not tridactyl's fault at least!

**Comment by vinc** [comment]
Same thing here but only after I updated Firefox from 72 to 82. It happens almost every time, I have to consciously wait about 1s before typing the url in a new tab or the text will disappear.

**Comment by bovine3dom** [comment]
If you're using about:blank you might as well use the Tridactyl builds that don't come with a new tab page : )

**Comment by vinc** [comment]
Thanks, I think it's a good solution while we wait until the bug is fixed in Firefox :+1:

**Comment by getzze** [comment]
I had the same problem and using the last https://tridactyl.cmcaine.co.uk/betas/nonewtab/ solved it :+1: 
Do you have any plan to submit the nonewtab version to the Firefox addon page, it took me ages to find this version and solve this long standing issue?
Thanks for making this indispensable extension!

**Comment by bovine3dom** [comment]
No, sorry - updating the add-ons on the AMO is a manual process and one I dislike enough already without having to do it twice for every release :)

I believe we already link to it from the new tab page.  I'm hoping that Mozilla will fix the issue eventually.

**Comment by braham-snyder** [comment]
If you'd like this Firefox bug fixed, you might want to vote for it. You can sign in using your github account. Then just open the "Details" dropdown on the bug's page to vote: https://bugzilla.mozilla.org/show_bug.cgi?id=1427535

(All credit to @maricn for finding it.)

<br><br>

Also, for what it's worth, this link directly downloads the latest "no new tab" build: https://tridactyl.cmcaine.co.uk/betas/nonewtab/tridactyl_no_new_tab_beta-latest.xpi

> I believe we already link to it from the new tab page.

You do link to it.

**Comment by bovine3dom** [comment]
@maricn my point with using the t / S binds is that you shouldn't need to use the Firefox address bar for very much.

With #3196 I think the reasons for using the address bar have got even fewer, since muscle memory now works pretty well with the Tridactyl commandline, provided that you allow a small amount of time (milliseonds) for the command line to open before sending it any further keys.

**Comment by maricn** [comment]
> @maricn my point with using the t / S binds is that you shouldn't need to use the Firefox address bar for very much.

Firefox address bar uses a familiar Firefox search/matching algorithm, and Tridactyl uses a different one. Often accessed websites are prioritized in Firefox address bar, and you can type what you want and then remember that you have it in the bookmarks, and just add an asterisk `*` to filter out results from history. Also any part of any word in the title of the target page can be typed in arbitrary order. There's more, and I've already voted for these two existing related issues - https://github.com/tridactyl/tridactyl/issues/936 and https://github.com/tridactyl/tridactyl/issues/1049.

> With #3196 I think the reasons for using the address bar have got even fewer, since muscle memory now works pretty well with the Tridactyl commandline, provided that you allow a small amount of time (milliseonds) for the command line to open before sending it any further keys.

I really appreciate the effort put into improving Tridactyl responsiveness and I notice it in my daily use. I'd update the original post to reflect that using t/S won't make the problem go away, but would somewhat improve the chances of avoiding the problem.

And as always, many thanks for the work on Tridactyl! <3

**Comment by bovine3dom** [comment]
I think you misunderstood me. I wasn't ever trying to suggest that `t<C-l>` would fix the problem - I was trying to say that `t[whatever you were going to use the address bar for]` would avoid the problem. : )

If you (perfectly reasonably) prefer the Firefox address bar then, yes, it will give you a different set of problems...

And thank you for your kind words!
